src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -0,0 +1,106 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Health.Fhir.TemplateManagement.Exceptions;
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Client
+{
+    // Todo need to be async
+
+    public class OrasClient : IOrasClient
+    {
+        private readonly string _imageReference;
+
+        public OrasClient(string imageReference)
+        {
+            EnsureArg.IsNotNull(imageReference, nameof(imageReference));
+
+            _imageReference = imageReference;
+        }
+
+        public async Task<bool> PullImageAsync(string outputFolder)
+        {
+            string command = $"pull  {_imageReference} -o {outputFolder}";
+            await OrasExecutionAsync(command, ".");
+            return true;
+        }
+
+        public async Task<bool> PushImageAsync(string inputFolder)
+        {
+            string argument = string.Empty;
+            string command = $"push {_imageReference}";
+
+            if (!Directory.Exists(inputFolder))
+            {
+                Console.WriteLine($"No file for push.");
+                return false;
+            }
+
+            var filePathToPush = Directory.EnumerateFiles(inputFolder, "*.tar.gz", SearchOption.AllDirectories);
+            foreach (var filePath in filePathToPush)
+            {
+                argument += $" {Path.GetRelativePath(inputFolder, filePath)}";
+            }
+
+            if (string.IsNullOrEmpty(argument))
+            {
```

Comment:
We can check whether the EnumeratedFileSet is empty instead of checking string here.

Reply:
Seems the same?